### ISSUE TITLE
consistent CLI interface, ie `-config=`

### ISSUE THREAD

**Issue by rjhornsby** [hashicorp/vault#14917]
#### Feature Description

Consul and vault agents' install process (manual or rpm) is nearly identical. So close that I ended up writing a single chef resource that can do both. The differences between installing on linux vs windows are much bigger than that of consul vs vault.

In the process, I realized that consul expects `-config-file=/path/to/some/file` and vault wants `-config=/path/to/some/file`. These flags mean the same thing, but have different names. To be fair, this is easy to work around in chef.

It's not a huge deal, but generally speaking it would be a nice-to-have if they were able to present the same interface when the options overlap like this.

It might be that consul aliases `-config=` or vault aliases `-config-file=` for a few versions until one form or the other can be reasonably deprecated.

#### Use Case(s)

General interface usability / consistency between tools. Specifically automation or scripting like puppet/chef/etc that is capable of managing multiple HashiCorp agents. It also helps humans not have to remember two different syntaxes.

**Comment by Amier3** [comment]
@rjhornsby 

I agree, `-config-file=` is just slightly more clear. I went ahead and transferred this to the vault repo so their team could chime in too 👍

**Comment by maxb** [comment]
I think Consul should be the one to change here, and `-config` should be the standard, because:

* The computer is quite capable of working out whether a passed path is a file or directory by itself, it doesn't need the human operator to tell it, and the CLI is more concise that way.
* Consul is the odd one out here, as Vault, Nomad and Boundary all implement `-config`.

**Comment by aphorise** [comment]
I agree with @maxb  - Consul is the odd one here and it would be great if an alias `-config` can be added to it to align with the rest of the products.

hey @amirakha1 @kschoche @ushakkodali  any idea how the may be realised? - it seems like a trivial ask - although in all fairness I've never reviewed or looked at the Consul code.

**Comment by rjhornsby** [comment]
If consul were to merge `-config-file` and `-config-dir` into/in favor of a single, unified `-config` I think that would make sense. Remedying consul from being the odd man out against the other HC tools is a reasonable approach.

As noted, the computer can certainly figure out if a path is a directory or a file, but it's not as easy for a human to look at `-config` and `-config-dir` and figure out why they both exist, and what the difference is.

The point being, if `-config-dir` goes away, I have no objection to `-config`

**Comment by aphorise** [comment]
@rjhornsby to be frank deprecating `-config-dir` from Consul would be a challenge and what's more as it stands in Vault we have:

 > [-config](https://www.vaultproject.io/docs/commands/server#config) (string: "") - Path to a configuration file or directory of configuration files. This flag can be specified multiple times to load multiple configurations. If the path is a directory, all files which end in .hcl or .json are loaded.

Which is implicitly the same as [Consuls `-config-dir`](https://www.consul.io/docs/agent/config/cli-flags#_config_dir). IMO a reasonable compromise to achieve what you're asking would be to have Consul adopt `-config`.
